packages/react-composites/review/react-composites.api.md

```diff
@@ -753,6 +770,62 @@ export type TopicChangedListener = (event: {
     topic: string;
 }) => void;
 
+// @beta (undocumented)
+export const UPLOAD_CANCELLED_EVENT = "uploadCancelled";
+
+// @beta (undocumented)
+export const UPLOAD_COMPLETED_EVENT = "uploadCompleted";
+
+// @beta (undocumented)
+export const UPLOAD_FAILED_EVENT = "uploadFailed";
+
+// @beta (undocumented)
+export const UPLOAD_PROGRESSED_EVENT = "uploadProgressed";
+
+// @beta
+export type UploadCanceledListener = () => void;
+
+// @beta
+export type UploadCompleteListener = (metaData: FileMetaData) => void;
+
+// @beta
+export class UploadedFile implements UploadEvents {
+    constructor(file: File);
+    cancelUpload(): void;
+    completeUpload(metaData: FileMetaData): void;
+    extension(): string;
```

Comment:
should extension should be a property?

Reply:
Going forward we might need to do some non trivial logic for getting file extensions. Hence, the method. Moreover, it will be used internally.